presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -3714,6 +3714,29 @@ public void testWindowsConstantExpression()
     }
 
     @Test
+    public void testDependentWindows()
+            throws Exception
+    {
+        // For such query as below generated plan has two adjacent window nodes where second depends on output of first.
+
+        MaterializedResult actual = computeActual("WITH " +
+                "t1 AS (" +
+                "SELECT extendedprice FROM lineitem ORDER BY orderkey, partkey LIMIT 2)," +
+                "t2 AS (" +
+                "SELECT extendedprice, sum(extendedprice) OVER() AS x FROM t1)," +
+                "t3 AS (" +
+                "SELECT max(x) OVER() FROM t2) " +
+                "SELECT * FROM t3");
+
+        MaterializedResult expected = resultBuilder(getSession(), DOUBLE)
```

Comment:
why not to use `assertQuery` instead?

`assertQuery(sql, "VALUES 59645.36, 59645.36");`